src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -267,6 +272,24 @@ public override BoundNode VisitCall(BoundCall node)
             return rewrittenBoundCall;
         }
 
+        private static MethodSymbol AdjustMethodForBaseInterfaceCall(BoundExpression receiverOpt, MethodSymbol method)
+        {
+            if (receiverOpt != null && receiverOpt.Kind == BoundKind.BaseReference)
+            {
+                var baseReference = (BoundBaseReference)receiverOpt;
+
+                if (method.IsImplementableInterfaceMember() &&
+                    baseReference.ExplicitBaseReferenceOpt?.Type.IsInterfaceType() == true)
+                {
+                    method = TypeSymbol.TryFindBaseImplementationInInterface((NamedTypeSymbol)baseReference.ExplicitBaseReferenceOpt.Type, method,
```

Comment:
>TryFindBaseImplementationInInterface [](start = 40, length = 36)

Didn't we do this during binding?  It feels like we might have been able to store the result at that time, to ensure the semantics are preserved rather than recomputed.

Reply:
> Didn't we do this during binding?  It feels like we might have been able to store the result at that time, to ensure the semantics are preserved rather than recomputed.

I do not believe re-computing is going to create a problem, we are re-computing accessors for properties and events in general and do not store them in the bound tree. This is not different, just one more step in re-computation. 

---
In reply to: [255799212](https://github.com/dotnet/roslyn/pull/33281#discussion_r255799212) [](ancestors = 255799212)